text/0000-experimental-coroutines.md

```diff
@@ -0,0 +1,471 @@
+- Feature Name: `coroutines`
+- Start Date: 2017-06-15
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This is an **experimental RFC** for adding a new feature to the language,
+coroutines (also commonly referred to as generators). This RFC is intended to be
+relatively lightweight and bikeshed free as it will be followed by a separate
+RFC in the future for stabilization of this language feature. The intention here
+is to make sure everyone's on board with the *general idea* of
+coroutines/generators being added to the Rust compiler and available for use on
+the nightly channel.
+
+# Motivation
+[motivation]: #motivation
+
+One of Rust's 2017 roadmap goals is ["Rust should be well-equipped for writing
+robust, high-scale servers"][goal]. A [recent survey][survey] has shown that
+the biggest blocker to robust, high-scale servers is ergonomic usage of async
```

Comment:
> e.g. at least 64 instead of 4KB RSS per connection on Linux)

This is less an issue of threads, and more an issue of glibc. Musl, in particular, [supports vastly smaller stacks than glibc does.](http://wiki.musl-libc.org/wiki/Functional_differences_from_glibc#Thread_stack_size)